File: codewars/6kyu/detect-pangram/dduck-detect-pangram.py
# https://www.codewars.com/kata/545cedaa9943f7fe7b000048/

from unittest import TestCase
import re

is_pangram = lambda s: len(set(re.compile('[a-z]').findall(s.lower()))) == 26

pangram = "The quick, brown fox jumps over the lazy dog!"
TestCase().assertEqual(is_pangram(pangram), True)

File: codewars/6kyu/detect-pangram/dduck-detect-pangram.js
// https://www.codewars.com/kata/545cedaa9943f7fe7b000048/train/javascript

const isPangram = (string) => new Set(string.toLowerCase().match(/[a-z]/gi)).size === 26;

console.log(isPangram('The quick brown fox jumps over the lazy dog.'), true);
console.log(isPangram('This is not a pangram.'), false);
